pkg/splunk/enterprise/indexercluster_test.go

```diff
@@ -145,6 +141,81 @@ func indexerClusterPodManagerTester(t *testing.T, method string, mockHandlers []
 			return c
 		},
 	}
+	return mgr, mockSplunkClient
+}
+
+// indexerClusterpodManagerVerifyRFPeersTester is used to verify replicas against RF using a indexerClusterPodManager
+func indexerClusterPodManagerVerifyRFPeersTester(t *testing.T, method string, mgr *indexerClusterPodManager,
+	desiredReplicas int32, wantPhase splcommon.Phase, wantCalls map[string][]spltest.MockFuncCall, wantError error) {
+
+	// initialize client
+	c := spltest.NewMockClient()
```

Comment:
We could assign the mock client in the **getIndexerClusterPodManagerAndClient** API? So that we don't have to assign again here.

Reply:
I could have done that, but if you assign it in **getIndexerClusterPodManagerAndClient** and then initialize `mgr` like this - 
```// initialize client
	client := spltest.NewMockClient()

	mgr := &indexerClusterPodManager{
		log:     scopedLog,
		cr:      &cr,
		secrets: secrets,
		newSplunkClient: func(managementURI, username, password string) *splclient.SplunkClient {
			c := splclient.NewSplunkClient(managementURI, username, password)
			c.Client = mockSplunkClient
			return c
		},
		c: client,
	}

then, you wont be able to call `c.CheckCalls(t, method, wantCalls)` at line 166, since `CheckCalls` is part of `spltest.MockClient` but `mgr.c` is of type `splcommon.ControllerClient`. Hence I think I am keeping it this way for now.